src/torchmetrics/functional/detection/panoptic_quality.py

```diff
@@ -267,6 +283,51 @@ def _panoptic_quality_update(
     return iou_sum, true_positives, false_positives, false_negatives
 
 
+def _panoptic_quality_update(
+    flatten_preds: Tensor,
+    flatten_target: Tensor,
+    cat_id_to_continuous_id: Dict[int, int],
+    void_color: Tuple[int, int],
+) -> Tuple[Tensor, Tensor, Tensor, Tensor]:
+    """Calculate stat scores required to compute accuracy.
+
+    Computed scores: iou sum, true positives, false positives, false negatives.
+
+    Args:
+        flatten_preds: A flattened prediction tensor, shape (B, num_points, 2).
+        flatten_target: A flattened target tensor, shape (B, num_points, 2).
+        cat_id_to_continuous_id: Mapping from original category IDs to continuous IDs.
+        void_color: an additional, unused color.
+
+    Returns:
+        - IOU Sum
+        - True positives
+        - False positives
+        - False negatives
+    """
+    device = flatten_preds.device
+    n_categories = len(cat_id_to_continuous_id)
+    iou_sum = torch.zeros(n_categories, dtype=torch.double, device=device)
+    true_positives = torch.zeros(n_categories, dtype=torch.int, device=device)
+    false_positives = torch.zeros(n_categories, dtype=torch.int, device=device)
+    false_negatives = torch.zeros(n_categories, dtype=torch.int, device=device)
+
+    # Loop over each sample independently: segments must not be matched across frames.
```

Comment:
I wish there was a nicer vectorised implementation, but couldn't think of anything meaningful so far (was thinking about hashing the batch_idx inside the instance_id, but that's a hack)

Reply:
yeah, it is not great to have for-loops inside the update method but I neither see an easy way of vectorizing this